src/cluster_asm.c

```diff
@@ -365,6 +368,153 @@ void asmTaskFree(asmTask *task) {
     zfree(task);
 }
 
+/* Convert the task state to the corresponding event. */
+int asmTaskStateToEvent(asmTask *task) {
+    if (task->operation == ASM_IMPORT) {
+        if (task->state == ASM_DONE) return ASM_EVENT_IMPORT_COMPLETED;
+        else if (task->state == ASM_FAILED) return ASM_EVENT_IMPORT_FAILED;
+        else return ASM_EVENT_IMPORT_STARTED;
+    } else {
+        if (task->state == ASM_DONE) return ASM_EVENT_MIGRATE_COMPLETED;
+        else if (task->state == ASM_FAILED) return ASM_EVENT_MIGRATE_FAILED;
+        else return ASM_EVENT_MIGRATE_STARTED;
+    }
+}
+
+/* Serialize ASM task information into a string for transmission to slaves.
+ * Format: "task_id:source_node:dest_node:operation:state:slot_ranges"
+ * Where slot_ranges is in the format "1000-2000 3000-4000 ..." */
+sds asmTaskSerialize(asmTask *task) {
+    sds serialized = sdsempty();
+
+    /* Add task ID */
+    serialized = sdscatprintf(serialized, "%s:", task->id);
+
+    /* Add source node ID (40 chars) */
+    serialized = sdscatlen(serialized, task->source, CLUSTER_NAMELEN);
+    serialized = sdscat(serialized, ":");
+
+    /* Add destination node ID (40 chars) */
+    serialized = sdscatlen(serialized, task->dest, CLUSTER_NAMELEN);
+    serialized = sdscat(serialized, ":");
+
+    /* Add operation type */
+    serialized = sdscatprintf(serialized, "%s:", task->operation == ASM_IMPORT ?
+                                                 "import" : "migrate");
+
+    /* Add current state */
+    serialized = sdscatprintf(serialized, "%s:", asmTaskStateToString(task->state));
+
+    /* Add slot ranges sds */
+    sds slot_ranges_str = slotRangeArrayToString(task->slot_ranges);
+    serialized = sdscatprintf(serialized, "%s", slot_ranges_str);
+    sdsfree(slot_ranges_str);
+
+    return serialized;
+}
+
+/* Deserialize ASM task information from a string and create a complete asmTask.
+ * Format: "task_id:source_node:dest_node:operation:state:slot_ranges"
+ * Returns a new asmTask on success, NULL on failure. */
+asmTask *asmTaskDeserialize(sds data) {
+    int count, idx = 0;
+    asmTask *task = NULL;
+    if (!data) return NULL;
+
+    sds *parts = sdssplitlen(data, sdslen(data), ":", 1, &count);
+    if (count < 6) goto err;
+
+    /* Parse task ID */
+    if (sdslen(parts[idx]) != CLUSTER_NAMELEN) goto err;
```

Comment:
I think we don't have a hard requirement on task id length. e.g. I'm not sure what does cluster plugin gives us currently. Maybe we should avoid length check?